acme/acme/crypto_util.py

```diff
@@ -0,0 +1,146 @@
+"""Crypto utilities."""
+import contextlib
+import logging
+import socket
+import sys
+
+from six.moves import range  # pylint: disable=import-error,redefined-builtin
+
+import OpenSSL
+
+from acme import errors
+
+
+logger = logging.getLogger(__name__)
+
+
+_DEFAULT_SSL_METHOD = OpenSSL.SSL.SSLv23_METHOD
```

Comment:
I can't think of any SSL vulnerabilities that would affect DVSNI challenge certificate serving / probing. It seems that SNI is all that matters in this challenge and the only attack surface is DOS, since prober needs to check certificate for expected contents anyway; SSL/TLS method used doesn't influence it.

`TLSv1_METHOD` is only compatible with `TLSv1_METHOD`, while `SSLv23_METHOD` is compatible with all other methods, including `TLSv2_METHOD`. https://www.openssl.org/docs/ssl/SSLv23_method.html. Code would have to be rewritten to attempt multiple methods before failing.

If I'm not wrong in the above statement, I would suggest to fix the constant name to `_DEFAULT_DVSNI_SSL_METHOD` and leave the comment inline in the source code.